src/main/java/org/jboss/ejb/protocol/remote/EJBClientChannel.java

```diff
@@ -664,7 +664,7 @@ private Marshaller getMarshaller() throws IOException {
     static IoFuture<EJBClientChannel> construct(final Channel channel, final DiscoveredNodeRegistry discoveredNodeRegistry, RetryExecutorWrapper retryExecutorWrapper) {
         FutureResult<EJBClientChannel> futureResult = new FutureResult<>();
         // now perform opening negotiation: receive server greeting
-        channel.receiveMessage(new Channel.Receiver() {
+        channel.receiveMessage(new TCCLConfiguringChannelReceiver(new Channel.Receiver() {
```

Comment:
Since these run per-request, it would be better to have a static method which configures the TCCL, and update the implementations of the various receivers to call the method rather than allocating yet another heap object.

Reply:
You are right of course - fixed accordingly.